optuna/storages/_rdb/storage.py

```diff
@@ -446,46 +468,50 @@ def _create_new_trial(
 
         # Retry a couple of times. Deadlocks may occur in distributed environments.
         n_retries = 0
-        with _create_scoped_session(self.scoped_session) as session:
-            while True:
+        MAX_RETRIES = 5
+        while n_retries <= MAX_RETRIES:
```

Comment:
We can also code like this:

```python
for _ in range(MAX_RETRIES):
    with ... as session:
        try:
            trial = ...
            return _create_frozen_trial_from_trial(...)
        except ...:
            raise
else:
    raise RuntimeError(...)
```
Instead of the current while loop.

Reply:
Agreed.